File: src/Edge.java
import java.util.*;

public class Edge implements Comparable<Edge> {
    private int endPoint1;
    private int endPoint2;
    
    public Edge(int a, int b) {
        this.endPoint1 = a > b? a:b;
        this.endPoint2 = a > b? b:a;
    }

    public int get1() {
        return this.endPoint1;
    }

    public int get2() {
        return this.endPoint2;
    }

    public int compareTo(Edge e2) {
        if(this.get1() < e2.get1() || this.get1() == e2.get1() && this.get2() < e2.get2()) {
            return -1;
        } else if (this.equals(e2)) {
            return 0;
        } else {
            return 1;
        }

    }

    public boolean equals(Edge e2) {
        return (this.get1() == e2.get1()) && (this.get2() == e2.get2());
    }

}

File: src/main/scala/de/sciss/tsp/Edge.scala
package de.sciss.tsp

object Edge {
  def apply(a: Int, b: Int): Edge = {
    val endPoint1 = if (a > b) a else b
    val endPoint2 = if (a > b) b else a
    new Edge(_1 = endPoint1, _2 = endPoint2)
  }
}
final case class Edge private(_1: Int, _2: Int) /*extends Comparable[Edge]*/ {
//  assert (_1 >= _2)

/*
  def compareTo(that: Edge): Int =
    if (this._1 < that._1 || this._1 == that._1 && this._2 < that._2) -1
    else if (this == that) 0
    else 1
*/
}
